src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,257 @@
+
+if TYPE_CHECKING:
+    from ansys.simai.core.data.projects import Project
+
+
+@dataclass
+class GlobalCoefficientUnit:
+    """Single Global Coefficient field.
+
+    Args:
+        formula: the Global Coefficient formula.
+        name: the name of the Global Coefficient.
+    """
+
+    formula: str
+    name: str
+
+
+@dataclass
+class ModelInput:
+    """The inputs of a model.
+
+    Args:
+        surface: the input surface variables.
+        boundary_conditions: the boundary conditions.
+    """
+
+    surface: list[str] = None
+    boundary_conditions: list[str] = None
+
+
+@dataclass
+class ModelOutput:
+    """The outputs of a model.
+
+    Args:
+        surface: the output surface variables.
+        volume: the output volume variables.
+    """
+
+    surface: list[str] = None
+    volume: list[str] = None
+
+
+@dataclass
+class ModelConfiguration:
+    """Configures the build of a model.
+
+    Args:
+        project: the project of the configuration.
+        build_preset: indicates the build duration. Available options:
+
+                    | *debug*: < 30 min
+
+                    | *short*: < 24 hours
+
+                    | *default*: < 2 days, default value.
+
+                    | *long*: < 1 week
+        continuous: indicates if continuous learning is enabled. Default is False.
+        input: the inputs of the model.
+        output: the outputs of the model.
+        global_coefficients: the Global Coefficients of the model.
+
+    Example:
+        Define a new configuration and launch a build.
+
+        .. code-block:: python
+
+            import ansys.simai.core as asc
+            from ansys.simai.core.data.model_configuration import (
+                ModelConfiguration,
+                ModelInput,
+                ModelOutput,
+            )
+
+            simai = asc.from_config()
+
+            # Get the project of interest
+            aero_dyn_project = simai.projects.get(name="aero-dyn")
+
+            # Define the inputs of the model
+            model_input = ModelInput(surface=["Velocity"], boundary_conditions=["Vx"])
+
+            # Define the outputs of the model
+            model_output = ModelOutput(
+                surface=["Pressure", "WallShearStress_0"], volume=["Velocity_0", "Pressure"]
+            )
+
+            # Define the model coefficient
+            global_coefficients = [("max(Pressure)", "maxpress")]
+
+            # Define the build configuration for the model
+            new_conf = ModelConfiguration(
+                project=aero_dyn_project,
+                build_preset="debug",
+                continuous=False,
+                input=model_input,
+                output=model_output,
+                global_coefficients=global_coefficients,
+                simulation_volume=simulation_volume,
+            )
+
+            # Launch a mode build with the new configuration
+            new_model = simai.models.build(new_conf)
+    """
+
+    project: "Project" = None
```

Comment:
```suggestion
    project: "Project"
```
Or, but I don't think we want that:
```suggestion
    project: "Optional[Project]" = None
```